docs/providers/aws/guide/plugins.md

```diff
@@ -510,6 +510,34 @@ This way, if user sets `anotherProp` by mistake to `some-string`, the Framework
 Serverless: Configuration error: functions.someFunc.events[0].yourPluginEvent.anotherProp should be number
 ```
 
+#### `defineFunctionEventProperties` helper
+
+When your plugin extend other plugin events definition for a specific provider, you can use the `defineFunctionEventProperties` to extend event definition with your custom properties.
+
+For exemple, if your plugin adds support to a new `documentation` property on `http` event from `aws` provider, you should add validations rules inside your plugin constructor for this new property.
+
+```javascript
+class NewEventPlugin {
+  constructor(serverless) {
+    this.serverless = serverless;
+
+    // Create schema for your properties. For reference use https://github.com/ajv-validator/ajv
+    serverless.configSchemaHandler.defineFunctionEventProperties('aws', 'http', {
+      properties: {
+        documentation: { type: 'string' },
+      },
+      required: ['documentation'],
+    });
+  }
+}
+```
+
+This way, if user sets `documentation` by mistake to `5`, the Framework would display an error:
```

Comment:
This statement is actually not true, as we're doing coercion `5` will be accepted as `"5"` :)

Bullet proof example would be to for example use object where primitive is expected and other way